backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/NativeInlineFunctionResolver.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.kotlin.backend.konan.lower
+
+import org.jetbrains.kotlin.backend.common.lower.*
+import org.jetbrains.kotlin.backend.common.lower.inline.DefaultInlineFunctionResolver
+import org.jetbrains.kotlin.backend.common.lower.inline.LocalClassesExtractionFromInlineFunctionsLowering
+import org.jetbrains.kotlin.backend.common.lower.inline.LocalClassesInInlineFunctionsLowering
+import org.jetbrains.kotlin.backend.common.lower.inline.LocalClassesInInlineLambdasLowering
+import org.jetbrains.kotlin.backend.common.runPostfix
+import org.jetbrains.kotlin.backend.konan.Context
+import org.jetbrains.kotlin.descriptors.impl.EmptyPackageFragmentDescriptor
+import org.jetbrains.kotlin.ir.declarations.IrDeclaration
+import org.jetbrains.kotlin.ir.declarations.IrFile
+import org.jetbrains.kotlin.ir.declarations.IrFunction
+import org.jetbrains.kotlin.ir.declarations.impl.IrFileImpl
+import org.jetbrains.kotlin.ir.symbols.IrFunctionSymbol
+import org.jetbrains.kotlin.ir.util.addChild
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.name.Name
+
+// TODO: This is a bit hacky. Think about adopting persistent IR ideas.
+internal class NativeInlineFunctionResolver(override val context: Context) : DefaultInlineFunctionResolver(context) {
+    override fun getFunctionDeclaration(symbol: IrFunctionSymbol): IrFunction {
+        val function = super.getFunctionDeclaration(symbol)
+        val body = function.body ?: return function
+
+        if (function in context.specialDeclarationsFactory.loweredInlineFunctions)
+            return function
+
+        context.specialDeclarationsFactory.loweredInlineFunctions.add(function)
+
+        PreInlineLowering(context).lower(body, function)
+
+        ArrayConstructorLowering(context).lower(body, function)
+
+        NullableFieldsForLateinitCreationLowering(context)
+                .runPostfix(true).transformFlat(function)
+        NullableFieldsDeclarationLowering(context)
+                .runPostfix(true).transformFlat(function)
```

Comment:
I noticed the difference between `runPostfix(true)` here and `runPostfix(false)` when applying these lowerings in main pipeline.
Apparently we have a bug in the latter case:
```kotlin
fun main() {
    val o = object : Any() {
        lateinit var x: Any
    }

    println(o.x)
}
```
Prints `null` on 1.4-M1 and master, fails with exception on 1.3.71.

Reply:
Could you please also check other lowerings for such issues?

Reply:
Other possible candidates are inheritors of `DeclarationTransformer`. We use `StripTypeAliasLowering` - for it doesn't matter since type aliases can only be on top level; `NullableFieldsForLateinitCreationLowering` and `NullableFieldsDeclarationLowering` - both need fixing; and the last are `DefaultArgumentStubGenerator` and `DefaultParameterCleaner` - they are run after the local declarations lowering, and therefore it is safe to skip local declarations at that point.